Make responses save upon clicking "Back" button

DoxasticFox opened the pull request:
Resolves #46.

In addition to doing as it says in the title, this PR changes the behaviour of the consent form somewhat to fix a UX bug I noticed while writing a new Cucumber scenario. Steps to reproduce the bug are as follows:

1. On step _n_ in the consent form, enter responses.
2. Click "Next" to navigate to step _n+1_. (Aside from the obvious effect of taking you to the next step, clicking "Next" will cause the responses to start being saved. The user can still navigate freely from step to step before the responses have finished saving.)
3. Before the responses have finished saving, click "Back" to return to step _n_. Because the form had not finished saving, it will load and display stale responses, or the default responses if none had previously ever finished saving.

To fix this issue, I prevented the user from navigating while responses are being saved. I did this by adding a transparent overlay with a spinner in the centre:

![Screenshot_20220829_122012](https://user-images.githubusercontent.com/11529449/187110316-6ba2f0e2-b0ea-4c6b-988e-5fe7413b3d00.png)

![Peek 2022-08-29 12-19](https://user-images.githubusercontent.com/11529449/187110309-914da6dc-bfa2-434c-83ab-d35a5762d3b7.gif)

mathob reviewed (approved):
I was unable to reproduce the error but I believe it could happen. The changes here work as described and when going through the steps I see the spinner momentarily preventing use of the Back button.

DoxasticFox commented:
@mathob I think it'd be practically impossible to reproduce manually without testing it over a high-latency network (e.g. the internet). I only noticed the bug because the cucumber tests click through the UI fast enough to reproduce it, and therefore failed after I added the new test until I added a spinner.
